src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php

```diff
@@ -53,14 +53,14 @@ class Ajax extends \Magento\Framework\App\Action\Action
      * @param \Magento\Framework\App\Action\Context                   $context            Controller action context.
      * @param \Magento\Framework\Controller\Result\JsonFactory        $jsonResultFactory  JSON result factory.
      * @param \Magento\Catalog\Model\Layer\Resolver                   $layerResolver      Layer resolver.
-     * @param \Magento\Catalog\Api\CategoryRepositoryInterfaceFactory $categoryRepository Category factory.
+     * @param \Magento\Catalog\Api\CategoryRepositoryInterface        $categoryRepository Category Repository.
      * @param \Magento\Catalog\Model\Layer\FilterList[]               $filterListPool     Filter list pool.
      */
     public function __construct(
         \Magento\Framework\App\Action\Context $context,
         \Magento\Framework\Controller\Result\JsonFactory $jsonResultFactory,
         \Magento\Catalog\Model\Layer\Resolver $layerResolver,
-        \Magento\Catalog\Api\CategoryRepositoryInterfaceFactory $categoryRepository,
```

Comment:
inject shared instance of the category repository so we can benefice from entities loaded within the registry from the layer